pkgs/build-support/php/hooks/default.nix

```diff
@@ -0,0 +1,26 @@
+{ buildPackages
+, callPackage
+, php
+, jq
+, lib
+, makeSetupHook
+, stdenvNoCC
+, unzip
+, xz
+, git
+}:
+
+{
+  composerSetupHook = makeSetupHook {
+      name = "composer-setup-hook.sh";
+      propagatedBuildInputs = [ php unzip xz git ];
```

Comment:
Do we need `git`? I'm guessing it's because one of the derivation requires a git repository as a source but this is going to be tricky to manage.

In the cache, it's a bare repository and in the `vendor/` directory we have the non-bare git repository. Both of them are not deterministic so they cannot be used directly in the FOD.

A first step could be not support lockfiles needing dependencies using source repositories and to document it as a known limitation. This is probably good enough for the majority of the use cases.